### PR TITLE
[Keyboard] AEBoards Constellation

### PR DESCRIPTION
**Pull request by Xelus22**
## Description

Add Constellation keyboard.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Add Constellation support

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/aeboards/constellation/readme.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+CONSTELLATION
+===
+
+A gasket 65% keyboard by [aeboards](https://aeboards.com/)
+
+Keyboard Maintainer: [Xelus22](https://github.com/Xelus22)  
+Hardware Supported: CONSTELLATION
+Hardware Availability: Custom keyboard group buys
```
Comment (fauxpark) [inline keyboards/aeboards/constellation/readme.md:8]:
```suggestion
# CONSTELLATION

A gasket 65% keyboard by [aeboards](https://aeboards.com/)

* Keyboard Maintainer: [Xelus22](https://github.com/Xelus22)
* Hardware Supported: CONSTELLATION
* Hardware Availability: Custom keyboard group buys
```

File: keyboards/aeboards/constellation/rules.mk [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# MCU name
+MCU = atmega32u4
+
+BOOTLOADER = atmel-dfu
```
Comment (fauxpark) [inline keyboards/aeboards/constellation/rules.mk:4]:
```suggestion
# Bootloader selection
BOOTLOADER = atmel-dfu
```

File: keyboards/aeboards/constellation/rules.mk [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality on B7 by default
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output on port C
+
```
Comment (fauxpark) [inline keyboards/aeboards/constellation/rules.mk:21]:
```suggestion
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = yes        # Console for debug
COMMAND_ENABLE = yes        # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/aeboards/constellation/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,21 @@
+#include QMK_KEYBOARD_H
```
Comment (noroadsleft) [inline keyboards/aeboards/constellation/keymaps/default/keymap.c:1]: Need a license header here.

File: keyboards/aeboards/constellation/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,37 @@
+#include QMK_KEYBOARD_H
```
Comment (noroadsleft) [inline keyboards/aeboards/constellation/keymaps/via/keymap.c:1]: Need a license header here.

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**